docs/supportedsites.rst

```diff
@@ -108,7 +108,7 @@ Tsumino              https://www.tsumino.com/            Galleries, Search Resul
 Tumblr               https://www.tumblr.com/             Images from Users, Likes, Posts, Tag-Searches      Optional (OAuth)
 Twitter              https://twitter.com/                Media Timelines, Timelines, Tweets                 Optional
 VSCO                 https://vsco.co/                    Images from Users, Collections, individual Images
-Wallhaven            https://wallhaven.cc/               individual Images, Search Results
+Wallhaven            https://wallhaven.cc/               individual Images, Search Results                  Optional
```

Comment:
`docs/supportedsites.rst` is auto-generated by `scripts/supportedsites.py`.
If anthing, you need to add a line there and run the script.
https://github.com/mikf/gallery-dl/blob/dc73d02d8780529b6c75822c38e8fa7759ceeb1d/scripts/supportedsites.py#L125-L126

And `Optional` without any additional information means it supports logging in with username and password, which it doesn't. It's more consistent, I think, to add `(API Key)` and link to the config option:

```
    "wallhaven"  : "Optional (`API Key <configuration.rst#extractorwallhavenapi-key>`__)"  # noqa
```